File: reverse.java
import java.util.Scanner;

public class ReverseParagraph {
    public static String reverseWord(String word) {
        if (word.length() == 0 || word.length() == 1) {
            return word;
        } else {
            return reverseWord(word.substring(1)) + word.charAt(0);
        }
    }

    public static String reverseParagraph(String paragraph) {
        String[] words = paragraph.split("\\s+");
        StringBuilder reversedParagraph = new StringBuilder();
        for (int i = words.length - 1; i >= 0; i--) {
            reversedParagraph.append(reverseWord(words[i])).append(" ");
        }
        return reversedParagraph.toString().trim();
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter a paragraph: ");
        String inputParagraph = scanner.nextLine();
        String reversedParagraph = reverseParagraph(inputParagraph);
        System.out.println("Reversed paragraph:");
        System.out.println(reversedParagraph);
    }
}

File: ASSIGNMENT_01/reverse.py
def reverse_word(word):
    if len(word) == 0 or len(word) == 1:
        return word
    else:
        return reverse_word(word[1:]) + word[0]

def reverse_paragraph(paragraph):
    words = paragraph.split()
    reversed_paragraph = " ".join(list(map(reverse_word, reversed(words))))
    return reversed_paragraph

input_paragraph = input("Enter a paragraph: ")
reversed_paragraph = reverse_paragraph(input_paragraph)
print("Reversed paragraph:")
print(reversed_paragraph)
